### PR TITLE
Throw an IllegalArgumentError error for mkdir('')

### PR DESCRIPTION
**Pull request by ben-albrecht**
Previously, `FileSystem.mkdir('')` resulted in a seg fault. Now it results in this message:

```
FileNotFoundError: No such file or directory (mkdir called with illegal path: '')
```

### CODE REVIEW

File: test/library/standard/FileSystem/mkdir/illegalArgs.good [hunk 1]
```diff
@@ -0,0 +1,3 @@
+IllegalArgumentError: mkdir called with illegal path: ''
```
Comment (lydia-duncan) [inline test/library/standard/FileSystem/mkdir/illegalArgs.good]: I don't see a .chpl file?

Comment (ben-albrecht) [reply test/library/standard/FileSystem/mkdir/illegalArgs.good]: Too late, @benharsh already approved.

**Comment by cassella** [comment]
A space character is a valid filename, at least on Linux..

**Comment by ben-albrecht** [comment]
> A space character is a valid filename, at least on Linux..

Removed that check entirely. Thanks @cassella!

>  isn't there somewhere else that would handle other kinds of invalid directory names? Would that be a better place to add this kind of check?

I didn't see any general "is this a valid filename?" function anywhere.

It looks like the runtime functions in `chpl-file-utils.c` sometimes do some argument checking, but generally rely on the system calls to emit system errors if something goes wrong.

**Comment by ben-albrecht** [comment]
It looks like `IO.open()` uses this pattern:

```chpl
  if (path == "") then
    try ioerror(ENOENT:syserr, "in open: path is the empty string");
```

File: test/library/standard/FileSystem/mkdir/illegalArgs.good [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FileNotFoundError: No such file or directory (mkdir called with illegal path: '')
```
Comment (benharsh) [inline test/library/standard/FileSystem/mkdir/illegalArgs.good]: Does this .good file need to be updated to say "SystemError"?

Comment (ben-albrecht) [reply test/library/standard/FileSystem/mkdir/illegalArgs.good]: Nope. The `ioerror(ENOENT:syserr)` throws a `FileNotFoundError`, which is just a subtype of `SystemError`.

**Pull request merged**